cmd/kube-apiserver/app/aggregator.go

```diff
@@ -132,9 +133,11 @@ func createAggregatorServer(aggregatorConfig *aggregatorapiserver.Config, delega
 	crdRegistrationController := crdregistration.NewAutoRegistrationController(
 		apiExtensionInformers.Apiextensions().InternalVersion().CustomResourceDefinitions(),
 		autoRegistrationController)
+	openapiController := openapi.NewController(apiExtensionInformers.Apiextensions().InternalVersion().CustomResourceDefinitions())
```

Comment:
odd to see an apiextensions-apiserver controller to be created here and not in apiextensions-apiserver itself.

Reply:
It's for addressing the one-minute aggregation latency we had in the [previous implementation](https://github.com/kubernetes/kubernetes/pull/67205). Please take a look at [this doc](https://docs.google.com/document/d/13lBj8Stdwku8BgL0fbT__4Iw97NRh77loJ_MoZuCGwQ/edit#) and see if other alternatives make more sense.

Reply:
Please add a comment making this clear, plus a TODO to fix it.

Reply:
What about having previous solution and a lighter version of this controller. What this controller does is to watch CRDs and any change to CRDs would result in a trigger in aggregation layer to check apiextension api-server for new spec. The  check is fast if the CRD change is not changing the spec (because of caching and etag). This way this controller is only acting as a performance booster rather than doing actual aggregation.

Reply:
The controller can be removed all together when we fixed the performance issue without any change in the apiextension api server and the CRDs implementation stays contained.

Reply:
Just saw this solution in your doc. let me think more. there maybe still a way to do this without race condition.

Reply:
yes ideally we want apiextension-apiserver to be well isolated, utilizing crd status to avoid race condition

I'd like to purpose some design which generalize apiservice api for crd and also utilize crd status, to do better in both isolation and performance. But since the isolation is not user-facing/urgent for the short future, having a TODO to move this PR forward sounds good to me. 

Will discuss with @mbohlool about plans next

Reply:
we (with @roycaihw ) discussed this issue and we think keeping the aggregation code inside extension apiserver is the most forward compatible solution (meaning any improvement to the performance would mostly happens in the aggregation layer and keeping this change in another layer will not affect that). To fix the performance issue, we can just set the resync duration to a small number (e.g 1 second) for delegation targets. These targets and in process http-handlers that are basically function calls and with etag, the call is very cheap when there is no change. 

We also discussed future improvements to the aggregation layer for this and @roycaihw is going to start a document on that. One good solution comes to my mind for that is to have a streaming/watch support for /openapi/v2 endpoint but we can discuss all solutions in that doc when it is ready.